fuel-vm/src/interpreter/gas/tests.rs

```diff
@@ -0,0 +1,108 @@
+use super::*;
+use test_case::test_case;
+
+struct GasChargeInput {
+    cgas: u64,
+    ggas: u64,
+    gas: u64,
+}
+#[derive(Debug, PartialEq, Eq)]
+struct GasChargeOutput {
+    cgas: u64,
+    ggas: u64,
+}
+#[test_case(GasChargeInput{cgas: 0, ggas: 0, gas: 0} => Ok(GasChargeOutput{ cgas: 0, ggas: 0}); "zero")]
+#[test_case(GasChargeInput{cgas: 0, ggas: 0, gas: 1} => Err(RuntimeError::Recoverable(PanicReason::OutOfGas)); "no gas")]
+// Currently panics
+// #[test_case(GasChargeInput{cgas: 2, ggas: 0, gas: 1} => Err(RuntimeError::Recoverable(PanicReason::OutOfGas)); "no global gas")]
```

Comment:
Why does this panic? Should we have a `TODO` comment here?

Reply:
It panics when the `optimized` feature flag is set. Which I don't know how to disable just this test case when that flag is set

Reply:
I don't think the optimized-feature should change the behavior, maybe this is a bug?

Anyway, you could do

```rust
#[cfg_attr(
    not(feature = "optimized"),
    test_case(GasChargeInput{cgas: 2, ggas: 0, gas: 1} => Err(RuntimeError::Recoverable(PanicReason::OutOfGas)); "no global gas")
)]
```

Reply:
It causes panics when they are errors without the flag

Reply:
Global gas can't be less than context gas. I added the handling of this case with the corresponding error.